### PR TITLE
fix(compass-schema): fix map rendering + add e2e tests COMPASS-6131

### PR DESCRIPTION
**Pull request by addaleax**
### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [x] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-schema/src/components/field/field.jsx [hunk 1]
```diff
@@ -127,7 +127,7 @@ class Field extends Component {
   getSemanticType(type) {
     // check if the type represents geo coordinates, if privacy settings allow
 
-    if (this.enableMaps) {
+    if (this.state.enableMaps) {
```
Comment (alenakhineika) [inline packages/compass-schema/src/components/field/field.jsx:130]: 👀

**Review by alenakhineika — APPROVED** [review]
Thx for finding and fixing it!

File: packages/compass-schema/src/components/field/field.jsx (hunk 1 above)
Comment (addaleax) [reply packages/compass-schema/src/components/field/field.jsx:130]: Let’s add this to the long list of “TypeScript would have caught this” :smile:

Comment (alenakhineika) [reply packages/compass-schema/src/components/field/field.jsx:130]: 💯

**Pull request merged**